### PR TITLE
Prepare all applications before activating when bootstrapping config server, take 2 [run-systemtest]

### PR DESCRIPTION
**Pull request by hmusum**
Same changes as in https://github.com/vespa-engine/vespa/pull/18720, but add back old code and use feature flag to switch between old and new way of redeploying applications. 

https://github.com/vespa-engine/vespa/pull/18737 will hopefull help with the issues seen when https://github.com/vespa-engine/vespa/pull/18720 was tried.

Please review only.

### CODE REVIEW

File: configserver/src/main/java/com/yahoo/vespa/config/server/ConfigServerBootstrap.java [hunk 1]
```diff
@@ -100,6 +129,8 @@ private ConfigServerBootstrap(ApplicationRepository applicationRepository, RpcSe
         this.sleepTimeWhenRedeployingFails = Duration.ofSeconds(configserverConfig.sleepTimeWhenRedeployingFails());
         this.exitIfRedeployingApplicationsFails = exitIfRedeployingApplicationsFails;
         rpcServerExecutor = Executors.newSingleThreadExecutor(new DaemonThreadFactory("config server RPC server"));
+        prepareAllAppsBeforeActivating = Flags.PREPARE_ALL_APPS_BEFORE_ACTIVATING_AT_BOOTSTRAP
+                .bindTo(flagSource).with(FetchVector.Dimension.ZONE_ID, applicationRepository.zone().toString());
```
Comment (hakonhall) [inline configserver/src/main/java/com/yahoo/vespa/config/server/ConfigServerBootstrap.java:133]: There is seldom any need to specify the zone dimension as all flags can be overridden per zone.  (The only reason I can think of is in the controller, and if it wants to do things differently per zone.)

**Review by hakonhall — COMMENTED** [review]
How would prepare-at-bootstrap work when bootstrapping 3 downed config server?  Each of the cfg would wait on the prepare (and activate) barriers, and is that OK before the config servers have completed the bootstrap?  

And in particular one would be faced with cfg1 trying to prepare application A that cfg2 has not prepared yet, and would cfg2 then be able to complete the prepare barrier for A?

**Comment by hmusum** [comment]
Redeplopying applications only happens when there is a version upgrade (version is persisted and read when server is started). So bootstrapping with the version that was last used should work as before.

File: configserver/src/main/java/com/yahoo/vespa/config/server/ConfigServerBootstrap.java (hunk 1 above)
Comment (hmusum) [reply configserver/src/main/java/com/yahoo/vespa/config/server/ConfigServerBootstrap.java:133]: Good point, will remove zone dimension.

**Review by hakonhall — APPROVED** [review]
What about the zone bootstrap case?

**Comment by hakonhall** [comment]
In the zone bootstrap case, there are no apps, so that's fine.

**Comment by hakonhall** [comment]
So what if all cfgs are down due to some bug or other, and to get it up we need to upgrade?

**Comment by baldersheim** [comment]
@hmusum and @hakonhall Is this PR still relevant ?

**Comment by hmusum** [comment]
Closing, bootstrap seems to work fine now and there are issues when it takes a long time from prepare to activate (new deployments might happen, then this needs to be retried etc.).

**Pull request closed without merging**